### DISCUSSION TITLE
🚨 looping through blade produces duplicates on alpine x-model array property

### DISCUSSION THREAD

**Discussion by marufdevops** [livewire/livewire#5370] [7. Bugs (archived)]
### Description:
when using alpine x-model with livewire it is appending duplicate value to alpine array when checked. 

### Implementation:

Here is a render method from component class: 
```php

protected $dueCommissions;

public function loadDueReward()
{
  //paginated query for $this->dueCommissions
}
public function render()
{
  //passing dueCommissions here, due to livewire limitation of handling paginated data
  return view('reward::livewire.due-reward', [
  'dueCommissions' => $this->dueCommissions
  ]);
}
```

```blade
//loading a fetching icon initially and re-rendering the actual data when ready with wire:init
<div class="main-content" wire:init='loadDueReward' wire:target='loadDueReward;gotoPage' x-data="{
    selectedUsers: []
}">
  @foreach ($dueCommissions ?? [] as $dueCommission)
    <tr wire:key='due-user-{{ $dueCommission->id }}'>
      <td class="table-cell-first">
        <input type="checkbox" class="form-checkbox" name="user" value="{{ $dueCommission->user_id }}" x-model="selectedUsers">
      </td>
    </tr>
  @endforeach
</div>
```
This code produce duplicates in the alpine selectedUsers when clicked
<img width="285" alt="Screen Shot 2022-11-18 at 6 44 04 PM" src="https://user-images.githubusercontent.com/56079318/202849191-3910cc50-5672-4bd1-95e1-a34adae85f50.png">

### Context 

- Livewire version: "livewire/livewire": "^2.10"
- Laravel version: "laravel/framework": "^9.0"
- alpinejs@3.8.1

**Comment by joshhanley** [comment]
@marufdevops please don't paste screenshots of code as it makes it difficult for us to assist. Instead paste the code directly in using code blocks.

When you paste code into github, you need to surround it in triple back ticks and then specify the file type like php or blade.

For example
\```php
\\ PHP code here
\```

\```blade
\\ HTML and blade code here
\```

Could you share your code which reproduces the issue with as little code as possible so we can investigate it?

Thanks!

**Reply by marufdevops** [reply]
Thank you for the tips. @joshhanley. Updated

**Reply by marufdevops** [reply]
Hey @joshhanley anything on this?

**Comment by marufdevops** [comment]
can you please help here @calebporzio. 

what it seems like if I remove the wire:init and call the function within the render method it works fine.

**Comment by joshhanley** [comment]
@marufdevops please be patient, you updated your discussion while it was the weekend for me.

To me it looks like an Alpine `x-model` issue as you are binding to the same property. Maybe try binding using the array index and seeing if that works?

Hope this helps!

**Reply by marufdevops** [reply]
ah sorry about that. didn't get what do you meant by binding to array index?

**Reply by joshhanley** [reply]
No worries 🙂

Have a look at this comment, it should show how to bind to an array. https://github.com/alpinejs/alpine/issues/144#issuecomment-667952544

**Reply by marufdevops** [reply]
ok I will try this, thanks for sharing. for now as a workaround what I did is removed the model binding, and pushing the item in array with click; marking them as check if found in array.
